servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLServerRequest.kt

```diff
@@ -0,0 +1,9 @@
+package com.expediagroup.graphql.server.execution
+
+import com.expediagroup.graphql.types.GraphQLRequest
+
+sealed class GraphQLServerRequest<T: Any>(val request: T)
+
+class GraphQLSingleRequest(graphQLRequest: GraphQLRequest) : GraphQLServerRequest<GraphQLRequest>(graphQLRequest)
```

Comment:
Should we move these to `graphql-kotlin-types` to be used by clients as well?

Reply:
I'm still playing around with the client batching but I don't think it will make much sense to do - I used sealed class so I can have single return type from the request parser method. While we could potentially follow the same on the client, I'm wondering if there is any benefit of using the extra wrapper over just

```kotlin
client.execute(request) // single
client.execute(listOf(request1, request2)) // batch

// or a vararg
client.execute(request1, request2)
```

Currently we also do not use the `GraphQLRequest` in the clients due to the difference in `variables` field definition - in server we want to access those as a generic map, on server we want to send `Any` (so it will serialize to a generic map).